### ISSUE TITLE
Libmamba fails when trying to use s3 conda channel

### ISSUE THREAD

**Issue by versci** [conda/conda#11412]
### Checklist

- [X] I added a descriptive title
- [X] I searched for other issues and couldn't find a solution or duplication
- [X] I looked at the [README.md](https://github.com/conda-incubator/conda-libmamba-solver) and didn't see anything to help

### What happened?

Libmamba fails when trying to access a Conda channel stored in a s3 object store. Note that the same command works fine when not using libmamba. The conda command is:
 
`conda env create --experimental-solver=libmamba --file environment.yml --name tmp3 --dry-run`

environment.yml defines several channels:

```
channels:
  - defaults
  - conda-forge
  - s3://gammaecho39/dfs/packages/conda/kdt/main
```

The error returned is:
```
RuntimeError: Download error (1) Unsupported protocol [s3://gammaecho39/dfs/packages/conda/kdt/main/noarch/repodata.json]
    Protocol "s3" not supported or disabled in libcurl
```

### Conda info

<details>
<summary><code>conda info</code></summary>

```Shell
active environment : base
    active env location : /GAAL/kisseri/conda_envs/miniconda
            shell level : 1
       user config file : /u1/kisseri/.condarc
 populated config files : /u1/kisseri/.condarc
          conda version : 4.12.0
    conda-build version : 3.21.8
         python version : 3.9.7.final.0
       virtual packages : __linux=3.10.0=0
                          __glibc=2.17=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /GAAL/kisseri/conda_envs/miniconda  (writable)
      conda av data dir : /GAAL/kisseri/conda_envs/miniconda/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /GAAL/kisseri/conda_envs/miniconda/pkgs
                          /u1/kisseri/.conda/pkgs
       envs directories : /GAAL/kisseri/conda_envs
                          /GAAL/kisseri/conda_envs/miniconda/envs
                          /u1/kisseri/.conda/envs
               platform : linux-64
             user-agent : conda/4.12.0 requests/2.27.1 CPython/3.9.7 Linux/3.10.0-1160.49.1.el7.x86_64 rhel/7.9 glibc/2.17
                UID:GID : 27795:40
             netrc file : None
           offline mode : False
```
</details>

### Conda config

```Shell
==> /u1/kisseri/.condarc <==
envs_dirs:
  - /GAAL/kisseri/conda_envs
```

### Conda list

<details>
<summary><code>conda list</code></summary>

```Shell
# packages in environment at /GAAL/kisseri/conda_envs/miniconda:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_openmp_mutex             4.5                       1_gnu    defaults
backports                 1.1                pyhd3eb1b0_0    defaults
backports.functools_lru_cache 1.6.4              pyhd3eb1b0_0    defaults
backports.tempfile        1.0                pyhd3eb1b0_1    defaults
backports.weakref         1.0.post1                  py_1    defaults
beautifulsoup4            4.10.0             pyh06a4308_0    defaults
boto3                     1.20.24            pyhd3eb1b0_0    defaults
botocore                  1.23.24            pyhd3eb1b0_0    defaults
brotlipy                  0.7.0           py39h27cfd23_1003    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
c-ares                    1.18.1               h7f8727e_0    defaults
ca-certificates           2022.3.29            h06a4308_0    defaults
certifi                   2021.10.8        py39h06a4308_2    defaults
cffi                      1.15.0           py39hd667e15_1    defaults
chardet                   4.0.0           py39h06a4308_1003    defaults
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
click                     8.0.4            py39h06a4308_0    defaults
conda                     4.12.0           py39h06a4308_0    defaults
conda-build               3.21.8           py39h06a4308_2    defaults
conda-content-trust       0.1.1              pyhd3eb1b0_0    defaults
conda-libmamba-solver     22.3.1             pyhd3eb1b0_0    defaults
conda-package-handling    1.7.3            py39h27cfd23_1    defaults
conda-verify              3.4.2                      py_1    defaults
cryptography              36.0.0           py39h9ce1e76_0    defaults
filelock                  3.6.0              pyhd3eb1b0_0    defaults
future                    0.18.2           py39h06a4308_1    defaults
glob2                     0.7                pyhd3eb1b0_0    defaults
icu                       58.2                 he6710b0_3    defaults
idna                      3.3                pyhd3eb1b0_0    defaults
importlib-metadata        4.11.3           py39h06a4308_0    defaults
importlib_metadata        4.11.3               hd3eb1b0_0    defaults
jinja2                    2.11.3             pyhd3eb1b0_0    defaults
jmespath                  0.10.0             pyhd3eb1b0_0    defaults
krb5                      1.19.2               hac12032_0    defaults
ld_impl_linux-64          2.35.1               h7274673_9    defaults
libarchive                3.4.2                h62408e4_0    defaults
libcurl                   7.82.0               h0b77cf5_0    defaults
libedit                   3.1.20210910         h7f8727e_0    defaults
libev                     4.33                 h7f8727e_1    defaults
libffi                    3.3                  he6710b0_2    defaults
libgcc-ng                 9.3.0               h5101ec6_17    defaults
libgomp                   9.3.0               h5101ec6_17    defaults
liblief                   0.11.5               h295c915_1    defaults
libmamba                  0.22.1               h1566912_0    defaults
libmambapy                0.22.1           py39hd09550d_0    defaults
libnghttp2                1.46.0               hce63b2e_0    defaults
libsolv                   0.7.20               h4ff587b_0    defaults
libssh2                   1.9.0                h1ba5d50_1    defaults
libstdcxx-ng              9.3.0               hd4cf53a_17    defaults
libxml2                   2.9.12               h03d6c58_0    defaults
lz4-c                     1.9.3                h295c915_1    defaults
markupsafe                2.0.1            py39h27cfd23_0    defaults
ncurses                   6.3                  h7f8727e_2    defaults
openssl                   1.1.1n               h7f8727e_0    defaults
patchelf                  0.13                 h295c915_0    defaults
pip                       21.2.4           py39h06a4308_0    defaults
pkginfo                   1.8.2              pyhd3eb1b0_0    defaults
psutil                    5.8.0            py39h27cfd23_1    defaults
py-lief                   0.11.5           py39h295c915_1    defaults
pycosat                   0.6.3            py39h27cfd23_0    defaults
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyopenssl                 21.0.0             pyhd3eb1b0_1    defaults
pysocks                   1.7.1            py39h06a4308_0    defaults
python                    3.9.7                h12debd9_1    defaults
python-dateutil           2.8.2              pyhd3eb1b0_0    defaults
python-libarchive-c       2.9                pyhd3eb1b0_1    defaults
pytz                      2021.3             pyhd3eb1b0_0    defaults
pyyaml                    6.0              py39h7f8727e_1    defaults
readline                  8.1.2                h7f8727e_1    defaults
reproc                    14.2.4               h295c915_1    defaults
reproc-cpp                14.2.4               h295c915_1    defaults
requests                  2.27.1             pyhd3eb1b0_0    defaults
ripgrep                   12.1.1                        0    defaults
ruamel_yaml               0.15.100         py39h27cfd23_0    defaults
s3transfer                0.5.0              pyhd3eb1b0_0    defaults
setuptools                58.0.4           py39h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_0    defaults
soupsieve                 2.3.1              pyhd3eb1b0_0    defaults
spdlog-fmt-embed          1.9.2                hd09550d_0    defaults
sqlite                    3.37.0               hc218d9a_0    defaults
tk                        8.6.11               h1ccaba5_0    defaults
tqdm                      4.62.3             pyhd3eb1b0_1    defaults
tzdata                    2021e                hda174b7_0    defaults
urllib3                   1.26.7             pyhd3eb1b0_0    defaults
wheel                     0.37.1             pyhd3eb1b0_0    defaults
xz                        5.2.5                h7b6447c_0    defaults
yaml                      0.2.5                h7b6447c_0    defaults
yaml-cpp                  0.7.0                h295c915_1    defaults
zipp                      3.7.0              pyhd3eb1b0_0    defaults
zlib                      1.2.11               h7f8727e_4    defaults
zstd                      1.4.9                haebb681_0    defaults
```
</details>

### Log file & additional context

```
>> conda env create --experimental-solver=libmamba --file environment.yml --name tmp3 --dry-run
***

NOTE: You are using the EXPERIMENTAL libmamba solver integration.

If something is not working as expected, please:

1. Go to https://github.com/conda/conda/issues/new/choose
2. Choose the "Libmamba Solver Feedback (Experimental Feature)" option
3. Attach the log file found in the following path:

/GAAL/kisseri/conda_envs/miniconda/pkgs/.logs/20220412-151235-801218.log

Thank you for your help!

***

conda_channel/linux-64                               6.3kB @ 554.2kB/s  0.0s
conda_channel/noarch                               127.0 B @  13.4kB/s  0.0s
pkgs/r/noarch                                                 No change
pkgs/r/linux-64                                               No change
pkgs/main/noarch                                   808.0kB @ 846.0kB/s  1.0s
pkgs/main/linux-64                                   4.4MB @   2.0MB/s  2.2s
conda-forge/noarch                                   7.9MB @   2.0MB/s  3.5s
[+] 6.1s
conda-forge/linux-64                 ━━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  20.2MB /  ??.?MB @   3.3MB/s                       6.1s
dfs/packages/conda/kdt/main/linux-64 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0.0 B /   1.0 B @  ??.?MB/s Unsupported protocol  6.0s
dfs/packages/conda/kdt/main/noarch   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0.0 B /   1.0 B @  ??.?MB/s Unsupported protocol  6.0sinfo     libmamba Download error (1) Unsupported protocol [s3://gammaecho39/dfs/packages/conda/kdt/main/linux-64/repodata.json]
    Protocol "s3" not supported or disabled in libcurl
info     libmamba Transfer done for 'dfs/packages/conda/kdt/main/linux-64'
info     libmamba Transfer finalized, status 0 [s3://gammaecho39/dfs/packages/conda/kdt/main/linux-64/repodata.json] 0 bytes
dfs/packages/conda/kdt/main/linux-64                ??.?MB @  ??.?MB/s             0 failed  6.1s
[+] 6.2s
conda-forge/linux-64               ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  20.4MB /  ??.?MB @   3.3MB/s                       6.2s
dfs/packages/conda/kdt/main/noarch ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0.0 B /   1.0 B @  ??.?MB/s Unsupported protocol  6.1sTraceback (most recent call last):
  File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/solver.py", line 224, in solve_final_state
    index = LibMambaIndexHelper(
  File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/solver.py", line 87, in __init__
    self._index = load_channels(
  File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/mamba_utils.py", line 117, in load_channels
    index = get_index(
  File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/mamba_utils.py", line 98, in get_index
    is_downloaded = dlist.download(api.MAMBA_DOWNLOAD_FAILFAST)
[+] 6.3s
conda-forge/linux-64               ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  20.4MB /  ??.?MB @   3.3MB/s                       6.3s
dfs/packages/conda/kdt/main/noarch ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0.0 B /   1.0 B @  ??.?MB/s Unsupported protocol  6.2s
# >>>>>>>>>>>>>>>>>>>>>> ERROR REPORT <<<<<<<<<<<<<<<<<<<<<<

    Traceback (most recent call last):
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/solver.py", line 224, in solve_final_state
        index = LibMambaIndexHelper(
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/solver.py", line 87, in __init__
        self._index = load_channels(
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/mamba_utils.py", line 117, in load_channels
        index = get_index(
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/mamba_utils.py", line 98, in get_index
        is_downloaded = dlist.download(api.MAMBA_DOWNLOAD_FAILFAST)
    RuntimeError: Download error (1) Unsupported protocol [s3://gammaecho39/dfs/packages/conda/kdt/main/noarch/repodata.json]
    Protocol "s3" not supported or disabled in libcurl

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda/exceptions.py", line 1082, in __call__
        return func(*args, **kwargs)
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_env/cli/main.py", line 80, in do_call
        exit_code = getattr(module, func_name)(args, parser)
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_env/cli/main_create.py", line 121, in execute
        solved_env = installer.dry_run(pkg_specs, args, env)
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_env/installers/conda.py", line 37, in dry_run
        pkgs = solver.solve_final_state()
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/solver.py", line 224, in solve_final_state
        index = LibMambaIndexHelper(
      File "/GAAL/kisseri/conda_envs/miniconda/lib/python3.9/site-packages/conda_libmamba_solver/utils.py", line 86, in __exit__
        raise exc_type(exc_value)
    RuntimeError: Download error (1) Unsupported protocol [s3://gammaecho39/dfs/packages/conda/kdt/main/noarch/repodata.json]
    Protocol "s3" not supported or disabled in libcurl

`$ /GAAL/kisseri/conda_envs/miniconda/bin/conda-env create --experimental-solver=libmamba --file environment.yml --name tmp3 --dry-run`

  environment variables:
                 CIO_TEST=<not set>
  CONDA_AUTO_UPDATE_CONDA=false
        CONDA_DEFAULT_ENV=base
                CONDA_EXE=/GAAL/kisseri/conda_envs/miniconda/bin/conda
             CONDA_PREFIX=/GAAL/kisseri/conda_envs/miniconda
    CONDA_PROMPT_MODIFIER=(base)
         CONDA_PYTHON_EXE=/GAAL/kisseri/conda_envs/miniconda/bin/python
               CONDA_ROOT=/GAAL/kisseri/conda_envs/miniconda
              CONDA_SHLVL=1
           CURL_CA_BUNDLE=<not set>
          LD_LIBRARY_PATH=/apps/qtrinst/lib/boost/sandbox:/apps/qtrinst/software/oracle/client/l
                          ib:/apps/qtrinst/software/mssql_x86_64/lib64:/apps/qtrinst/software/sy
                          base/OCS-15_0/lib3p64:/apps/qtrinst/software/sybase/OCS-15_0/lib:/apps
                          /qtrinst/software/R/lib64/R/lib:/apps/qtrinst/software/unixodbc/lib:/a
                          pps/qtrinst/software/perl/lib/site_perl/5.20.0::/nfsapps/qtrinst/softw
                          are/java/jre/lib/amd64:/nfsapps/qtrinst/software/java/jre/lib/amd64/se
                          rver:/usr/local/sybase/sybase15/ASIQ-15_3/IQ-15_3/lib64:/usr/local/syb
                          ase/sybase15/OCS-15_0/lib:/usr/local/sybase/sybase15/OCS-15_0/lib3p:/u
                          sr/local/sybase/sybase15/OCS-15_0/lib:/usr/local/sybase/sybase15/OCS-1
                          5_0/lib3p64:/usr/local/sybase/sybase15/OCS-15_0/lib3p:/nfsapps/OMD_cli
                          ent/onetick/Linux64_client/one_market_data/one_tick/bin
                  MANPATH=/u1/kisseri/.nvm/versions/node/v17.4.0/share/man::/opt/puppetlabs/pupp
                          et/share/man
                     PATH=/GAAL/kisseri/conda_envs/miniconda/bin:/u1/kisseri/.poetry/bin:/GAAL/k
                          isseri/conda_envs/miniconda/bin:/GAAL/kisseri/conda_envs/miniconda/con
                          dabin:/u1/kisseri/.nvm/versions/node/v17.4.0/bin:/GAAL/kisseri/apps/cm
                          ake/bin:/local/apps/autosys/iXp/bin:/apps/qtrinst/software/oracle/clie
                          nt/bin:/apps/qtrinst/software/mssql_x86_64/bin:/usr/local/sybase/sybas
                          e15/OCS-15_0/bin:/usr/local/sybase/sybase15/ASIQ-15_3/IQ-15_3/bin64:/u
                          sr/local/sybase/sybase15/ASIQ-15_3/DBISQL/bin:/apps/qtrinst/software/s
                          ybase/OCS-15_0/bin64:/apps/qtrinst/software/sybase/ASIQ-15_3/IQ-15_3/b
                          in64:/apps/qtrinst/software/sybase/ASIQ-15_3/DBISQL/bin:/apps/qtrinst/
                          software/R/bin:/apps/qtrinst/software/R/lib64/R/bin:/apps/qtrinst/soft
                          ware/python/bin:/apps/qtrinst/software/matlab/bin:/apps/qtrinst/softwa
                          re/unixodbc/bin:/apps/qtrinst/software/perl/bin:/apps/qtrinst/software
                          /sas/SASFoundation/9.4:/apps/qtrinst/software/sas/SASFoundation/9.4/bi
                          n:/apps/qtrinst/software/sas/SASFoundation/9.4/utilities/bin:/bin:/usr
                          /bin:/usr/local/bin:/apps/qtrinst/bin
           QT_PLUGIN_PATH=/usr/lib64/kde4/plugins:/usr/lib/kde4/plugins
       REQUESTS_CA_BUNDLE=<not set>
            SSL_CERT_FILE=<not set>
                all_proxy=<set>
                ftp_proxy=<set>
               http_proxy=<set>
              https_proxy=<set>
                 no_proxy=<set>

     active environment : base
    active env location : /GAAL/kisseri/conda_envs/miniconda
            shell level : 1
       user config file : /u1/kisseri/.condarc
 populated config files : /u1/kisseri/.condarc
          conda version : 4.12.0
    conda-build version : 3.21.8
         python version : 3.9.7.final.0
       virtual packages : __linux=3.10.0=0
                          __glibc=2.17=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /GAAL/kisseri/conda_envs/miniconda  (writable)
      conda av data dir : /GAAL/kisseri/conda_envs/miniconda/etc/conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /GAAL/kisseri/conda_envs/miniconda/pkgs
                          /u1/kisseri/.conda/pkgs
       envs directories : /GAAL/kisseri/conda_envs
                          /GAAL/kisseri/conda_envs/miniconda/envs
                          /u1/kisseri/.conda/envs
               platform : linux-64
             user-agent : conda/4.12.0 requests/2.27.1 CPython/3.9.7 Linux/3.10.0-1160.49.1.el7.x86_64 rhel/7.9 glibc/2.17
                UID:GID : 27795:40
             netrc file : None
           offline mode : False


An unexpected error has occurred. Conda has prepared the above report.

If submitted, this report will be used by core maintainers to improve
future releases of conda.
[+] 46.3s
conda-forge/linux-64               ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━╸━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  20.4MB /  ??.?MB @   3.3MB/s                      46.3s
dfs/packages/conda/kdt/main/noarch ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0.0 B /   1.0 B @  ??.?MB/s Unsupported protocol 46.2s
Timeout reached. No report sent.
```

**Comment by beeankha** [comment]
@jaimergp and @wolfv , this appears specific to `libmamba`'s implementation; are there any future plans to support an s3 protocol?

**Comment by jaimergp** [comment]
I feel this needs support in `libmamba`. I can reproduce with both `conda-libmamba-solver` and `mamba`. See this issue for more details: https://github.com/mamba-org/mamba/issues/394

That said, we could error out in a better way other than this long, unexpected report.

**Comment by wolfv** [comment]
@beeankha we will have support for S3 buckets once the powerloader integration is finished (https://github.com/mamba-org/powerloader) – that would be our new backend for all things downloading (and uploading).

**Comment by jezdez** [comment]
FWIW I merged https://github.com/conda-incubator/conda-libmamba-solver/pull/41.

**Comment by beeankha** [comment]
Hi @versci , could you please let us know if you're still encountering this issue?

**Comment by jaimergp** [comment]
A general [solution](https://github.com/conda-incubator/conda-libmamba-solver/pull/49) to "channels not supported by libmamba" will be available in conda-libmamba-solver 22.8, coming this week. We can close this issue.
